lib/checks/color/color-contrast-evaluate.js

```diff
@@ -34,21 +42,33 @@ function colorContrastEvaluate(node, options = {}, virtualNode) {
 		return undefined;
 	}
 
-	const noScroll = !!(options || {}).noScroll;
 	const bgNodes = [];
-	const bgColor = getBackgroundColor(node, bgNodes, noScroll);
-	const fgColor = getForegroundColor(node, noScroll, bgColor);
+	const bgColor = getBackgroundColor(node, bgNodes, false);
+	const fgColor = getForegroundColor(node, false, bgColor);
 
 	const nodeStyle = window.getComputedStyle(node);
 	const fontSize = parseFloat(nodeStyle.getPropertyValue('font-size'));
 	const fontWeight = nodeStyle.getPropertyValue('font-weight');
-	const bold = parseFloat(fontWeight) >= 700 || fontWeight === 'bold';
+	const bold = parseFloat(fontWeight) >= boldValue || fontWeight === 'bold';
+
+	const contrast = getContrast(bgColor, fgColor);
+	const ptSize = Math.ceil(fontSize * 72) / 96;
+	const isSmallFont =
+		(bold && ptSize < boldPt) || (!bold && ptSize < nonBoldPt);
+
+	const { expected, minThreshold, maxThreshold } = isSmallFont
+		? contrastRatio.normal
+		: contrastRatio.large;
+	const isValid = contrast > expected;
 
-	const cr = hasValidContrastRatio(bgColor, fgColor, fontSize, bold);
+	// ratio is outside range
+	if (contrast < minThreshold || contrast > maxThreshold) {
```

Comment:
I think it's better to make these optional:

```suggestion
	if (
		(typeof minThreshold === 'number' && contrast < minThreshold) || 
		(typeof maxThreshold === 'number' && contrast > maxThreshold)
	) {
```